File: part3-backend-challenge/src/main/java/com/payment/entity/Merchant.java
package com.payment.entity;

import io.micronaut.data.annotation.*;
import io.micronaut.data.model.DataType;
import io.micronaut.serde.annotation.Serdeable;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

@Serdeable
@MappedEntity(value = "merchants", schema = "merchants")
public class Merchant {

    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    private Integer merchantId; // SERIAL → Integer

    private String merchantName;

    private String merchantStatus;

    private String contactInfo;

    private String merchantCategory;

    private String merchantLocation;

    @DateCreated
    private Instant createdAt;

    @DateUpdated
    private Instant lastUpdated;

    private BigDecimal merchantRating;

    private Integer numOrders;

    private String paymentMethod;

    private String merchantLogo;

    private String merchantWebsite;

    private String merchantType;

    @MappedProperty(type = DataType.STRING_ARRAY)
    private List<String> merchantTags;

    private Integer page;

    private Integer itemsPerPage;

    private String errorMessage;

    private Boolean isLoading;

    // Constructor
    public Merchant() {
    }

    // Getters and Setters

    public Integer getMerchantId() {
        return merchantId;
    }

    public void setMerchantId(Integer merchantId) {
        this.merchantId = merchantId;
    }

    public String getMerchantName() {
        return merchantName;
    }

    public void setMerchantName(String merchantName) {
        this.merchantName = merchantName;
    }

    public String getMerchantStatus() {
        return merchantStatus;
    }

    public void setMerchantStatus(String merchantStatus) {
        this.merchantStatus = merchantStatus;
    }

    public String getContactInfo() {
        return contactInfo;
    }

    public void setContactInfo(String contactInfo) {
        this.contactInfo = contactInfo;
    }

    public String getMerchantCategory() {
        return merchantCategory;
    }

    public void setMerchantCategory(String merchantCategory) {
        this.merchantCategory = merchantCategory;
    }

    public String getMerchantLocation() {
        return merchantLocation;
    }

    public void setMerchantLocation(String merchantLocation) {
        this.merchantLocation = merchantLocation;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Instant createdAt) {
        this.createdAt = createdAt;
    }

    public Instant getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(Instant lastUpdated) {
        this.lastUpdated = lastUpdated;
    }

    public BigDecimal getMerchantRating() {
        return merchantRating;
    }

    public void setMerchantRating(BigDecimal merchantRating) {
        this.merchantRating = merchantRating;
    }

    public Integer getNumOrders() {
        return numOrders;
    }

    public void setNumOrders(Integer numOrders) {
        this.numOrders = numOrders;
    }

    public String getPaymentMethod() {
        return paymentMethod;
    }

    public void setPaymentMethod(String paymentMethod) {
        this.paymentMethod = paymentMethod;
    }

    public String getMerchantLogo() {
        return merchantLogo;
    }

    public void setMerchantLogo(String merchantLogo) {
        this.merchantLogo = merchantLogo;
    }

    public String getMerchantWebsite() {
        return merchantWebsite;
    }

    public void setMerchantWebsite(String merchantWebsite) {
        this.merchantWebsite = merchantWebsite;
    }

    public String getMerchantType() {
        return merchantType;
    }

    public void setMerchantType(String merchantType) {
        this.merchantType = merchantType;
    }

    public List<String> getMerchantTags() {
        return merchantTags;
    }

    public void setMerchantTags(List<String> merchantTags) {
        this.merchantTags = merchantTags;
    }

    public Integer getPage() {
        return page;
    }

    public void setPage(Integer page) {
        this.page = page;
    }

    public Integer getItemsPerPage() {
        return itemsPerPage;
    }

    public void setItemsPerPage(Integer itemsPerPage) {
        this.itemsPerPage = itemsPerPage;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public Boolean getIsLoading() {
        return isLoading;
    }

    public void setIsLoading(Boolean isLoading) {
        this.isLoading = isLoading;
    }
}

File: part4-frontend-challenge/src/types/merchant.ts
export interface Merchant {
    merchantId: number;
    merchantName: string;
    merchantStatus: 'Active' | 'Inactive';
    contactInfo: string;
    merchantCategory: string;
    merchantLocation: string;
    merchantRating: number;
    numOrders: number;
    paymentMethod: string;
    merchantLogo: string;
    merchantWebsite: string;
    merchantType: string;
}

export interface MerchantResponse {
    response_code: string;
    response_message: string;
    data: {
        merchants: Merchant[];
    };
}

export interface CreateMerchantPayload {
    merchantName: string;
    merchantStatus: 'Active' | 'Inactive' | 'Suspended';
    contactInfo: string;
    merchantCategory: string;
    merchantLocation: string;
    paymentMethod: string;
    merchantWebsite: string;
    merchantType: string;
    merchantTags: string[];
  }

  export interface CreateMerchantResponse {
    response_code: string;
    response_message: string;
    data: {
      merchantId: string;
      message: string;
    };
  }

  export interface UpdateMerchantPayload {
    merchantName: string;
    merchantStatus: "Active" | "Inactive" | "Suspended";
    contactInfo: string;
    merchantCategory: string;
    merchantLocation: string;
    merchantRating: number;
    numOrders: number;
    paymentMethod: string;
    merchantLogo: string;
    merchantWebsite: string;
    merchantType: "Individual" | "Business" | "Franchise";
    merchantTags: string[];
  }

  export interface UpdateMerchantResponse {
    response_code: string;
    response_message: string;
    data: {
      merchantId: string;
      message: string;
    };
  }


  export interface MerchantDetailResponse {
    merchantId: number;
    merchantName: string;
    merchantStatus: 'Active' | 'Inactive' | 'Suspended';
    contactInfo: string;
    merchantCategory: string;
    merchantLocation: string;
    merchantRating: number;
    numOrders: number;
    paymentMethod: string;
    merchantLogo: string;
    merchantWebsite: string;
    merchantType: 'Individual' | 'Business' | 'Franchise';
  }
  